### PR TITLE
⭐️  Add ms365.sharepointonline.site resource

### PR DESCRIPTION
**Pull request by HRouhani**

### CODE REVIEW

**Comment by HRouhani** [comment]
Screenshot from 2023-12-16 15-22-26

**Comment by HRouhani** [comment]
Screenshot from 2023-12-16 15-31-39

**Comment by HRouhani** [comment]
@preslavgerchev    my guess is that ConvertTo-Json   does not remove [] !

File: providers/ms365/resources/ms365.lr [hunk 1]
```diff
@@ -398,6 +398,16 @@ ms365.sharepointonline {
   spoTenant dict
   // SharePoint Online tenant sync client restriction
   spoTenantSyncClientRestriction dict
+  // SharePoint Online tenant sites
+  spoSites []ms365.sharepointonline.site
```
Comment (preslavgerchev) [inline providers/ms365/resources/ms365.lr:402]: @HRouhani i have made this to be a list, any reason why it cannot be?

File: providers/ms365/connection/sharepoint_report.go [hunk 2]
```diff
@@ -17,20 +17,22 @@ var sharepointReport = `
...
 Connect-PnPOnline -AccessToken $token -Url $url
 
 $SPOTenant = (Get-PnPTenant)
 $SPOTenantSyncClientRestriction = (Get-PnPTenantSyncClientRestriction)
+$SPOSite = (Get-PnPTenantSite)
 
 $sharepoint = New-Object PSObject
 Add-Member -InputObject $sharepoint -MemberType NoteProperty -Name SPOTenant -Value $SPOTenant
 Add-Member -InputObject $sharepoint -MemberType NoteProperty -Name SPOTenantSyncClientRestriction -Value $SPOTenantSyncClientRestriction
+Add-Member -InputObject $sharepoint -MemberType NoteProperty -Name SPOSite -Value $SPOSite
```
Comment (preslavgerchev) [inline providers/ms365/connection/sharepoint_report.go:31]: lets return all sites from the script. even it we dont end up using them its better to keep this script as simple as possible and do any other logic in the provider itself

**Comment by preslavgerchev** [comment]
@HRouhani i have made some changes so that we keep adding resources-only instead of dicts (specially for top-level stuff)

File: providers/ms365/connection/sharepoint_report.go (hunk 2 above)
Comment (HRouhani) [reply providers/ms365/connection/sharepoint_report.go:31]: awesome idea, Thanks for taking care of it

File: providers/ms365/resources/ms365.lr (hunk 1 above)
Comment (HRouhani) [reply providers/ms365/resources/ms365.lr:402]: Thanks, looks really good.

**Comment by HRouhani** [comment]
@preslavgerchev  The one which is need for now is ms365.sharepointonline.spoSites, which works as expected and therefor this PR can be merged. The one related to ms365.sharepointonline.site {*} is not working and from my perspective is not needed now. So you can improve and fix it whenever you got time. I just put it here for your info. Thanks for your Great work

**Comment by preslavgerchev** [comment]
> @preslavgerchev The one which is needed for now is ms365.sharepointonline.spoSites, which works as expected and therefor this PR can be merged. The one related to ms365.sharepointonline.site {*} is not working and from my perspective is not needed now. So you can improve and fix it whenever you got time. I just put it here for your info. Thanks for your Great work
> 
> Screenshot from 2023-12-18 10-59-07

`ms365.sharepointonline.site` is just the resource itself, we don't support this currently as it has no init function, only `ms365.sharepointonline.spoSites` is supported

**Review by imilchev — APPROVED** [review]

**Pull request merged**